deeplearning4j/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/main/java/org/deeplearning4j/models/embeddings/learning/impl/sequence/DBOW.java

```diff
@@ -222,12 +222,20 @@ public INDArray inferSequence(Sequence<T> sequence, long nextRandom, double lear
                 lookupTable.layerSize() + 1);
 
 
-        INDArray ret = Nd4j.createUninitialized(this.lookupTable.getWeights().dataType(),lookupTable.layerSize());
+
+        INDArray ret = Nd4j.createUninitializedDetached(this.lookupTable.getWeights().dataType(),lookupTable.layerSize());
         Nd4j.rand(ret,random);
         ret.subi(0.5).divi(lookupTable.layerSize());
         if(configuration.getWorkers() > 1) {
             Nd4j.getEnvironment().setMaxThreads(numThreadsOriginal);
         }
+
+        try {
+            random.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
This looks a bit out of place. Is this a left over from debugging?

That also applies to the other occurrences of this pattern in DM.java and SequenceVectors.java too.

Reply:
No this actually needs to be closed. We don't have a proper deallocatable for this. I discovered this using jemalloc/jeprof.

Reply:
I don't mean the closing, but the catching of every exception just to re-throw it.